google/cloud/BUILD.bazel

```diff
@@ -27,12 +27,32 @@ capture_build_info(
     ],
 )
 
+config_setting(
+    name = "have_open_telemetry",
+    flag_values = {
+        "//:experimental-open_telemetry": "true",
+        "@io_opentelemetry_cpp//api:with_abseil": "true",
+    },
+)
+
+config_setting(
+    name = "not_have_open_telemetry",
+    flag_values = {"//:experimental-open_telemetry": "false"},
+)
+
 load(":google_cloud_cpp_common.bzl", "google_cloud_cpp_common_hdrs", "google_cloud_cpp_common_srcs")
 
 cc_library(
     name = "google_cloud_cpp_common_private",
     srcs = google_cloud_cpp_common_srcs + ["internal/build_info.cc"],
     hdrs = google_cloud_cpp_common_hdrs,
+    target_compatible_with = select(
+        {
+            ":have_open_telemetry": [],
+            ":not_have_open_telemetry": [],
+        },
+        no_match_error = "OpenTelemetry must be built with Abseil. To fix this error, supply the `--@io_opentelemetry_cpp//api:with_abseil` flag to your build command.",
```

Comment:
Maybe:

```
no_match_error = """
You requested OpenTelemetry support for `google-cloud-cpp`, but OpenTelemetry is configured
without Abseil support.  This configuration will result in ODR violations, and possibly build-time
errors, as OpenTelemetry redefines Abseil symbols in this case, and `google-cloud-cpp` uses
Abseil directly too.

To fix this problem, supply the `--@io_opentelemetry_cpp//api:with_abseil` flag to your build
command, or set this value in your `bazelrc` file(s).
"""
```

Reply:
Thanks for the suggestion. It is more informative, than what I suggested.

```
$ bazel build //:common
Starting local Bazel server and connecting to it...
ERROR: .../google-cloud-cpp/google/cloud/BUILD.bazel:45:11: configurable attribute "target_compatible_with" in //google/cloud:google_cloud_cpp_common_private doesn't match this configuration: 


You requested OpenTelemetry support for `google-cloud-cpp`, but OpenTelemetry is
configured without Abseil support. This configuration will result in ODR
violations, and possibly build-time errors, as OpenTelemetry redefines Abseil
symbols in this case, and `google-cloud-cpp` uses Abseil directly too.

To fix this problem, supply the `--@io_opentelemetry_cpp//api:with_abseil` flag
to your build command, or set this value in your `.bazelrc` file(s).


This instance of //google/cloud:google_cloud_cpp_common_private has configuration identifier ab52c64. To inspect its configuration, run: bazel config ab52c64.

For more help, see https://docs.bazel.build/configurable-attributes.html#why-doesnt-my-select-choose-what-i-expect.

ERROR: Analysis of target '//:common' failed; build aborted: 
```